### PR TITLE
Tactical application of hystrix to S3Support

### PR DESCRIPTION
**Pull request by ajordens**
- Avoids a scenario where S3 is unavailable for brief periods of time

### CODE REVIEW

File: front50-web/src/main/groovy/com/netflix/spinnaker/front50/config/Front50WebConfig.groovy [hunk 1]
```diff
@@ -71,4 +77,27 @@ public class Front50WebConfig extends WebMvcConfigurerAdapter {
...
     return new ItemDAOHealthIndicator(itemDAO: pipelineStrategyDAO)
   }
+
+  @Bean
+  HystrixRuntimeExceptionHandler hystrixRuntimeExceptionHandler() {
+    return new HystrixRuntimeExceptionHandler()
+  }
+
+  @ControllerAdvice
+  static class HystrixRuntimeExceptionHandler {
```
Comment (ajordens) [inline front50-web/src/main/groovy/com/netflix/spinnaker/front50/config/Front50WebConfig.groovy]: Pretty close to what `igor`is doing

File: front50-s3/src/main/java/com/netflix/spinnaker/front50/model/S3Support.java [hunk 2]
```diff
@@ -103,20 +104,30 @@ public boolean isHealthy() {
...
-      T item = deserialize(s3Object);
-      item.setLastModified(s3Object.getObjectMetadata().getLastModified().getTime());
-      return item;
-    } catch (IOException e) {
-      throw new IllegalStateException(e);
-    } catch (AmazonS3Exception e) {
-      if (e.getStatusCode() == 404) {
-        throw new NotFoundException(String.format("No item found with id of %s", id.toLowerCase()));
-      }
+    return new SimpleHystrixCommand<T>(
```
Comment (ajordens) [inline front50-s3/src/main/java/com/netflix/spinnaker/front50/model/S3Support.java]: If there's an issue retrieving from S3, attempt to serve a previously cached value

**Comment by ajordens** [comment]
Had a situation earlier this week where S3 was returning errors ... this PR introduces hystrix with a fallback for `findbyId` that will return a recently cached value in the event that the S3 lookup fails.

Hystrix exports a set of default metrics which makes it a little nicer than the try/catch alternative.

In this particular case, the failure to read from S3 resulted in a handful of failed deploys due to `clouddriver` not being able to retrieve the applications `legacyUdf` preference.

**Comment by ajordens** [comment]
FYI @ewiseblatt 

PTAL @spinnaker/netflix-reviewers

**Comment by tomaslin** [comment]
LGTM

**Pull request merged**